packages/react/src/components/dialog/Dialog.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { ReactNode, ReactNodeArray, useEffect, RefObject } from 'react';
+import ReactDOM from 'react-dom';
+
+// import core base styles
+import 'hds-core';
+import styles from './Dialog.module.scss';
+import classNames from '../../utils/classNames';
+import { useTheme } from '../../hooks/useTheme';
+import { IconCross } from '../../icons';
+import { DialogActionButtons } from './dialogActionButtons/DialogActionButtons';
+import { DialogHeader } from './dialogHeader/DialogHeader';
+import { DialogContent } from './dialogContent/DialogContent';
+
+export interface DialogCustomTheme {
+  '--accent-line'?: string;
+  '--color-overlay'?: string;
+  '--width'?: string;
+}
+
+export type DialogProps = React.PropsWithChildren<{
```

Comment:
Is this React.PropsWithChildren necessary? I thought it helps in that we would not need to define the children prop ourself? But since we are defining it at line 56, I wonder if this adds value for us?

Reply:
I'll fix this.